src/bundle.c

```diff
@@ -723,6 +728,35 @@ static int install_bundles(struct list *bundles, struct list **subs, int current
 	to_install_files = filter_out_existing_files(to_install_files);
 	grabtime_stop(&times);
 
+	/* Step 1.5: Now that we know the bundle is valid, check if we have enough space */
+	ptr = list_head(to_install_bundles);
+	while (ptr) {
```

Comment:
You should actually skip this entire block if the user passes `--skip-diskspace-check` instead of calculating it every time regardless, *especially* since it requires recursing the manifest when we might not have the entire manifest tree on the system at this point (for example for bundles that are already installed on the system).

Reply:
@matthewrsj - Agreed, this was suggested above and I agree entirely and will change for V2.

@otaviobp - I like the for option, but it seems to me most of the code uses the while syntax. I may change this.

Reply:
I, for one, prefer the `while` loop because I think it's more readable but I know @icarus-sparry prefers the `for` loop. At the end of the day I'm fine with either, both are plenty readable.

Reply:
I much prefer the `for` style as well, and in at least one place it would have removed a bug if it had been used. The existing style is to use while loops when traversing linked lists, and for loops for index calculations, so whilst I hate it, I have given up on this fight.

Reply:
+1 for a LIST_FOREARCH macro